production/helm/loki/templates/bloom-builder/service-bloom-builder-headless.yaml

```diff
@@ -1,5 +1,5 @@
 {{- $isDistributed := eq (include "loki.deployment.isDistributed" .) "true" -}}
-{{- if $isDistributed -}}
+{{- if (and $isDistributed (gt (int .Values.bloomBuilder.replicas) 0)) -}}
```

Comment:
I think you need an `or .Values.bloomBuilder.autoscaling.enabled`  as well ... because we explicitly leave `replicas: 0` and set `autosacling.enabled` in our environment

Reply:
Good catch! Haven't thought about that.